R/module_teal.R

```diff
@@ -136,6 +138,27 @@ srv_teal <- function(id, modules, teal_data_rv, filter = teal_slices()) {
       title = "SessionInfo"
     )
 
+    output$lockFile <- downloadHandler(
+      filename = function() {
+        "renv.lock"
+      },
+      content = function(file) {
+        user_lockfile <- getOption("teal.renv.lockfile", "")
+        if (!file.exists(user_lockfile)) {
+          teal_lockfile <- getOption("teal.internal.renv.lockfile")
+          while (!file.exists(teal_lockfile)) {
+            logger::log_trace("lockfile not created yet, retrying...")
+            Sys.sleep(0.25)
+          }
```

Comment:
this one smells a little with infinite loop in case something unexpected happens for some reason
it would be great to introduce some limit on iterations (max_iter) or time (timeout)

Reply:
great idea!

Reply:
66709be18bbdf9575f2a1c89e73807637e1be9ef